spec/views/layouts/application.html.slim_spec.rb

```diff
@@ -5,10 +5,24 @@
 
   before do
     allow(view).to receive(:user_fully_authenticated?).and_return(true)
+    allow(view).to receive(:decorated_session).and_return(
+      DecoratedSession.new(sp: nil, view_context: nil).call
+    )
     allow(view.request).to receive(:original_url).and_return('http://test.host/foobar')
     allow(view).to receive(:current_user).and_return(User.new)
   end
 
+  context 'no content for nav present' do
+    it 'displays only the logo' do
+      render
+
+      expect(rendered).to have_xpath('//nav[contains(@class, "bg-light-blue")]')
+      expect(rendered).to_not have_link(t('shared.nav_auth.my_account'), href: profile_path)
+      expect(rendered).to_not have_content(t('shared.nav_auth.welcome'))
+      expect(rendered).to_not have_link(t('links.sign_out'), href: destroy_user_session_path)
+    end
+  end
+
```

Comment:
Maybe I missed it, but I don't see a test for the scenario where the user came from an SP. We could either add it here by stubbing `decorated_session` to return `ServiceProviderSessionDecorator`, or add some expectations in the loa1_sso_spec to make sure the profile links are not present.